pkg/mimirpb/compat_rw2_test.go

```diff
@@ -323,6 +365,31 @@ func TestRW2Unmarshal(t *testing.T) {
 		err = received.Unmarshal(data)
 		require.ErrorContains(t, err, "invalid")
 	})
+
+	t.Run("messages where the first symbol is empty string are rejected", func(t *testing.T) {
```

Comment:
Isn't the opposite? We error out if the first symbol is not an empty string.

```suggestion
	t.Run("messages where the first symbol is not empty string are rejected", func(t *testing.T) {
```